compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp

```diff
@@ -453,10 +453,18 @@ class TransposeReshapeGenericDotGeneral
             ? RankedTensorType::get(newShape, resultType.getElementType())
             : op.getType();
 
-    Value result = rewriter.create<mhlo::DotGeneralOp>(
-        op.getLoc(), newResultType, lhs, rhs, dimensionNumbers,
-        op.precision_configAttr());
-
+    auto newOp = rewriter.create<mhlo::DotGeneralOp>(op.getLoc(), newResultType,
+                                                     lhs, rhs, dimensionNumbers,
+                                                     op.precision_configAttr());
+
+    // Copy over unknown attributes as we currently rely on it to let user tune
+    // lowering parameters.
+    ArrayRef<StringRef> odsAttrs = op.getAttributeNames();
+    for (NamedAttribute kv : op->getAttrs())
+      if (!llvm::is_contained(odsAttrs, kv.getName().getValue()))
+        newOp->setAttr(kv.getName(), kv.getValue());
```

Comment:
style nit: IREE wraps single statements with braces

```suggestion
    ArrayRef<StringRef> odsAttrs = op.getAttributeNames();
    for (NamedAttribute kv : op->getAttrs()) {
      if (!llvm::is_contained(odsAttrs, kv.getName().getValue())) {
        newOp->setAttr(kv.getName(), kv.getValue());
      }
    }
```